core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -249,15 +257,37 @@ class OpenMPTargetExecTeamMember {
     // and L1 shmem size. TEAM_REDUCE_SIZE = 512 bytes saved per team for
     // hierarchical reduction. There is an additional 10% of the requested
     // scratch memory allocated per team as padding. Hence the product with 0.1.
+    //
+    // Use llvm extensions for dynamic shared memory with compilers/architecture
+    // combinations where it is supported.
+    //
+    // Size allocated in HBM will now change based on whether we use llvm
+    // extensions.
+#if defined(KOKKOS_IMPL_OPENMPTARGET_LLVM_EXTENSIONS)
```

Comment:
Are they other LLVM extensions that you might be willing to use?
Should we be more specific in the macro name?

Reply:
There are active plans towards working on extensions that might be considered "useful" for accelerators but not in OpenMP spec (for now at-least). Thats the reason I kept the macro name generic.  But there is none that can be used in near future.